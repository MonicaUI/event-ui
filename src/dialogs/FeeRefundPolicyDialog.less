@import (reference) './EventStatusDialog.less';

.popUpDialogContainer {
  .dialogContainer();
  max-height: 100vh;
}

.refundPolicyContainer {
  padding-bottom: 3rem;
}

.feeRefundSection {
  padding: 2.5rem 3rem 0 3rem;
}

.cancelledByHeader {
  margin: 1.75rem 0 1.25rem 0;

  &:after {
    content: '';
    display: table;
    clear: both;
  }

  .cancelledByHeaderText {
    float: left;
  }

  .cancelledByHeaderAmountText {
    float: right;
  }
}

.dateAmountWrapper {
  padding: 0 0.75rem 0.5rem 0.75rem;

  &:after {
    content: '';
    display: table;
    clear: both;
  }

  .cancelledByDate {
    float: left;
  }

  .amountRefunded {
    float: right;
  }
}
