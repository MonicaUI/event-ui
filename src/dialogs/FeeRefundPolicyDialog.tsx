import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { merge } from 'lodash';
import ThemeableComponent from 'nucleus-widgets/lib/ThemeableComponent';
import { openDialogContainer, closeDialogContainer } from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import { resolve } from '@cvent/nucleus-dynamic-css';
import getDialogContainerStyle from './shared/getDialogContainerStyle';
import DialogHeader from './shared/DialogHeader';
import EventStatusStyles from './EventStatusDialog.less';
import RefundPolicyStyles from './FeeRefundPolicyDialog.less';
import { dateInObjectFormat } from 'event-widgets/utils/feeUtils';
import { withStyles } from './ThemedDialog';

const FeeRefundPolicyDialogStyles = {
  classes: RefundPolicyStyles
};

/**
 * Dialog for displaying Fees Refund Policy.
 */
class FeeRefundPolicyDialog extends ThemeableComponent<$TSFixMe, $TSFixMe> {
  static propTypes = {
    translate: PropTypes.func,
    onClose: PropTypes.func,
    feeRefundData: PropTypes.object,
    eventCurrencyCode: PropTypes.object,
    translateCurrency: PropTypes.object,
    eventTimeZone: PropTypes.object
  };

  getElementBackground: $TSFixMe;
  getElementInlineStyle: $TSFixMe;
  props: $TSFixMe;

  getStyleObject() {
    return {
      dialogHeader: this.getElementInlineStyle('header'),
      dragContainer: this.getElementBackground('content1'),
      dragHandle: this.getElementBackground('content2'),
      text: this.getElementInlineStyle('text'),
      boldText: {
        ...this.getElementInlineStyle('text'),
        fontWeight: 700
      },
      cancelledByHeader: {
        ...this.getElementInlineStyle('text'),
        borderBottom: `1px solid ${this.getElementInlineStyle('text').color}`,
        padding: '0 0.75rem 0.25rem 0.75rem'
      },
      originalPrice: {
        ...this.getElementInlineStyle('text'),
        margin: '0.5rem 0 1.75rem'
      }
    };
  }

  /**
   * Returns 'refund amount' for a refund type and charge policy.
   */
  calculateRefundAmount = (refundPolicy, earlyBirdPrice) => {
    let calculatedRefundAmt;
    const refundType = refundPolicy.refundType;
    const maxRefund = earlyBirdPrice.maximumRefundAmount;
    const refundAmount = refundPolicy.amount;
    const feeAmount = earlyBirdPrice.amount;
    const percentage = refundPolicy.percentage;
    const cancellationAmount = refundPolicy.cancellationAmount;

    switch (refundType) {
      // Refund a fixed amount
      case 1: {
        calculatedRefundAmt = refundAmount;
        break;
      }

      // Charge a cancellation fee and refund the remaining amount paid
      case 3: {
        calculatedRefundAmt = feeAmount - cancellationAmount;
        break;
      }

      // Refund a percentage of the amount paid
      case 4: {
        calculatedRefundAmt = (percentage * feeAmount) / 100;
        break;
      }
      default:
        break;
    }
    if (calculatedRefundAmt < 0) {
      return 0;
    }
    if (maxRefund < calculatedRefundAmt) {
      return maxRefund;
    }
    return calculatedRefundAmt;
  };

  getChargePolicyLabeltext = (sortedChargePolicies, index) => {
    const { translate, eventCurrencyCode, translateCurrency, eventTimeZone } = this.props;
    const style = this.getStyleObject();
    const Styles = {
      ...FeeRefundPolicyDialogStyles,
      style
    };
    let title;
    const pricePaidLabel = (
      <div {...resolve(Styles, 'originalPrice')}>
        <span>
          {translate('EventWidgets_FeeWidget_RefundModal_PricePaid__resx', {
            pricePaid: `${translateCurrency(sortedChargePolicies[index].amount, eventCurrencyCode)}`
          })}
        </span>
      </div>
    );

    if (!sortedChargePolicies || sortedChargePolicies.length < 2) {
      if (sortedChargePolicies.length === 1) {
        return pricePaidLabel;
      }
      return null;
    } else if (sortedChargePolicies && index === sortedChargePolicies.length - 1) {
      const date = dateInObjectFormat(sortedChargePolicies[index - 1].effectiveUntil, eventTimeZone);
      title = (
        <div {...resolve(Styles, 'boldText')}>
          {translate('EventWidgets_FeeWidget_RefundModal_RefundAfterDate__resx', {
            day: translate(date.day),
            month: translate(date.month),
            date: date.date,
            year: date.year
          })}
        </div>
      );
    } else {
      const date = dateInObjectFormat(sortedChargePolicies[index].effectiveUntil, eventTimeZone);
      title = (
        <div {...resolve(Styles, 'boldText')}>
          {translate('EventWidgets_FeeWidget_RefundModal_RefundBeforeDate__resx', {
            day: translate(date.day),
            month: translate(date.month),
            date: date.date,
            year: date.year
          })}
        </div>
      );
    }

    return (
      <div>
        {title}
        {pricePaidLabel}
      </div>
    );
  };

  addForEachEarlyBirdPrice = feeRefundData => {
    const { translate } = this.props;
    const style = this.getStyleObject();
    const Styles = {
      ...FeeRefundPolicyDialogStyles,
      style
    };
    const sortedChargePolicies = Object.values(feeRefundData.chargePolicies).sort((a, b) => {
      // @ts-expect-error ts-migrate(2362) FIXME: The left-hand side of an arithmetic operation must... Remove this comment to see the full error message
      return new Date((a as $TSFixMe).effectiveUntil) - new Date((b as $TSFixMe).effectiveUntil);
    });
    return sortedChargePolicies.map((earlyBirdPrice, index) => {
      const prevEarlyBirdPrice = index > 0 ? sortedChargePolicies[index - 1] : null;
      const refundPolicyComp = this.addRefundPolicyComponent(
        prevEarlyBirdPrice,
        earlyBirdPrice,
        feeRefundData.refundPolicies
      );
      if (!refundPolicyComp || !refundPolicyComp.length) {
        return;
      }
      return (
        <div {...resolve(FeeRefundPolicyDialogStyles, 'feeRefundSection')} key={(earlyBirdPrice as $TSFixMe).id}>
          {this.getChargePolicyLabeltext(sortedChargePolicies, index)}
          <div {...resolve(Styles, 'cancelledByHeader')}>
            <span {...resolve(Styles, 'cancelledByHeaderText', 'text')}>
              {translate('EventWidgets_FeeWidget_RefundModal_CancelledBy__resx')}
            </span>
            <span {...resolve(Styles, 'cancelledByHeaderAmountText', 'text')}>
              {translate('EventWidgets_FeeWidget_RefundModal_AmountRefunded__resx')}
            </span>
          </div>
          <div>{refundPolicyComp}</div>
        </div>
      );
    });
  };

  addRefundPolicyComponent = (prevEarlyBirdPrice, earlyBirdPrice, refundPolicyData) => {
    const { translate, eventCurrencyCode, translateCurrency, eventTimeZone } = this.props;
    const style = this.getStyleObject();
    const Styles = {
      ...FeeRefundPolicyDialogStyles,
      style
    };

    const applicableRefundPolicies = Object.values(refundPolicyData)
      .filter(refundPolicy => {
        return prevEarlyBirdPrice
          ? // @ts-expect-error ts-migrate(2362) FIXME: The left-hand side of an arithmetic operation must... Remove this comment to see the full error message
            new Date((refundPolicy as $TSFixMe).effectiveUntil) - new Date(prevEarlyBirdPrice.effectiveUntil) > 0
          : true;
      })
      .sort((a, b) => {
        // @ts-expect-error ts-migrate(2362) FIXME: The left-hand side of an arithmetic operation must... Remove this comment to see the full error message
        return new Date((a as $TSFixMe).effectiveUntil) - new Date((b as $TSFixMe).effectiveUntil);
      });
    if (!applicableRefundPolicies || !applicableRefundPolicies.length) {
      return null;
    }
    return applicableRefundPolicies.map(refundPolicy => {
      const date = dateInObjectFormat((refundPolicy as $TSFixMe).effectiveUntil, eventTimeZone);
      return (
        <div {...resolve(FeeRefundPolicyDialogStyles, 'dateAmountWrapper')} key={(refundPolicy as $TSFixMe).id}>
          <div {...resolve(Styles, 'cancelledByDate', 'text')}>
            {translate('_dateLabels_day_month_year__resx', {
              day: translate(date.day),
              month: translate(date.month),
              date: date.date,
              year: date.year
            })}
          </div>
          <div {...resolve(Styles, 'amountRefunded', 'text')}>
            {`${translateCurrency(this.calculateRefundAmount(refundPolicy, earlyBirdPrice), eventCurrencyCode)}`}
          </div>
        </div>
      );
    });
  };

  addFeeRefundModelData = () => {
    const { feeRefundData } = this.props;

    return (
      <div {...resolve(FeeRefundPolicyDialogStyles, 'refundPolicyContainer')}>
        {this.addForEachEarlyBirdPrice(feeRefundData)}
      </div>
    );
  };

  render() {
    const {
      onClose,
      translate,
      feeRefundData: { feeName },
      style
    } = this.props;
    const dialogHeaderText = translate('EventWidgets_FeeWidget_RefundPolicyModalHeader__resx', {
      FeeName: translate(feeName)
    });
    return (
      <div>
        <DialogHeader
          text={dialogHeaderText}
          onClose={onClose}
          closeFallbackText="fallbackback"
          style={style}
          classes={EventStatusStyles}
        />
        {this.addFeeRefundModelData()}
      </div>
    );
  }
}

/**
 * Function to get the style object to pass into the share prompt dialog.
 */
function dialogStyle(globalTheme, sections) {
  return {
    ...globalTheme,
    header: { styleMapping: 'header3' },
    text: { styleMapping: 'body1' },
    content1: { ...merge({}, globalTheme, sections.content1), styleMapping: 'custom' },
    content2: { ...merge({}, globalTheme, sections.content2), styleMapping: 'custom' }
  };
}

const FeeRefundPolicyDialogWithStyles = withStyles(FeeRefundPolicyDialog);

/**
 * A connect wrapper around the Fee dialog to inject
 * the appropriate style information into its props.
 */
const ConnectedFeeRefundPolicyDialog = connect((state: $TSFixMe, props: $TSFixMe) => {
  const {
    website: {
      theme: { sections, global }
    },
    customFonts
  } = state;
  const style = merge({}, dialogStyle(global, sections), { customFonts });
  return {
    ...props.dialogConfig,
    style
  };
})(FeeRefundPolicyDialogWithStyles);

/**
 * Helper function to open the Fee refund policy dialog.
 */
export const openFeeRefundPolicyDialog = (feeObject: $TSFixMe) => {
  return (dispatch: $TSFixMe, getState: $TSFixMe): $TSFixMe => {
    const boundCloseDialog = () => dispatch(closeDialogContainer());
    const dialog = (
      <ConnectedFeeRefundPolicyDialog
        onClose={boundCloseDialog}
        translate={feeObject.translate}
        eventCurrencyCode={feeObject.eventCurrencyCode}
        translateCurrency={feeObject.translateCurrency}
        feeRefundData={feeObject.fee}
        eventTimeZone={feeObject.eventTimeZone}
      />
    );
    return dispatch(
      openDialogContainer(dialog, boundCloseDialog, {
        classes: {
          dialogContainer: (FeeRefundPolicyDialogStyles as $TSFixMe).popUpDialogContainer
        },
        style: {
          dragContainer: {
            ...getDialogContainerStyle(getState()),
            height: '50%'
          }
        }
      })
    );
  };
};
