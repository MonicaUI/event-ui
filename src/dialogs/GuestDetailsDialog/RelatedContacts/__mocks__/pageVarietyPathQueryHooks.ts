import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        guestRegistration {
          addGuestFromRelatedContacts {
            style {
              relatedContactsModalStyles
            }
            relatedContactsModalText
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useGuestRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
