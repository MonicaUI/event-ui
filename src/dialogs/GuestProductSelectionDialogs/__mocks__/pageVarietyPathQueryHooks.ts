import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        registration {
          admissionItems {
            display {
              fees
            }
          }
          sessions {
            display {
              fees
            }
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
