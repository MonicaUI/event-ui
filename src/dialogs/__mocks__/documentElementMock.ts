export const getMockedMessageContainer = (): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    <div name="messageContainer" tabIndex=-1>message container</div>
    </div>
  `;
  const mockMessageContainerFocus = jest.fn();
  const mockMessageContainer = document.getElementsByName('messageContainer')[0];
  mockMessageContainer.focus = mockMessageContainerFocus;
  return mockMessageContainer;
};

export const getMockedFieldInputs = (fieldNames: $TSFixMe): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    ${fieldNames.map(fieldName => `<input name=${fieldName}>input-${fieldName}</input>`)}
    </div>
  `;
  const mockInputFocus = jest.fn();
  const mockInputContainers = [];
  fieldNames.forEach(fieldName => {
    const mockInputContainer = document.getElementsByName(fieldName)[0];
    mockInputContainer.focus = mockInputFocus;
    mockInputContainers.push(mockInputContainer);
  });
  return mockInputContainers;
};
