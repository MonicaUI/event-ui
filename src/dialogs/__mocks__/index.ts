/* eslint-env jest */
import noopAction from '../../testUtils/noopAction';

export const openAlreadyRegisteredDialog = jest.fn(noopAction);

export const openDeclineRegistrationDialog = jest.fn(noopAction);

export const openStartNewRegistrationDialogFromPageLanding = jest.fn(noopAction);

export const openStartNewRegistrationDialogDuringRegistration = jest.fn(noopAction);

export const startNewRegistrationAndNavigateToRegistration = jest.fn(noopAction);

export const openEventWaitlistDialog = jest.fn(noopAction);

export const openEventTemporaryClosedErrorDialog = jest.fn(noopAction);

export const openNoAdmissionItemAvailableForRegistrationTypeDialog = jest.fn(noopAction);

export const openEventStatusDialog = jest.fn(noopAction);

export const openPrivateEventErrorDialog = jest.fn(noopAction);

export const openRegistrationInOtherTabDialog = jest.fn(noopAction);

export const openKnownErrorDialog = jest.fn(noopAction);

export const openPaymentProcessingErrorDialog = jest.fn(noopAction);

export const openInformationAlreadyUsedDialog = jest.fn(noopAction);

export const openPartialRegistrationConfirmationDialog = jest.fn(noopAction);

export const openPreviewModeWarningDialog = jest.fn(noopAction);

export const openCapacityReachedDialog = jest.fn(noopAction);

export const openGroupCancelRegistrationDialog = jest.fn(noopAction);

export const openSessionOverlapWarningDialog = jest.fn(noopAction);

export const openGroupMemberRemoveDialog = jest.fn(noopAction);

export const openSharePromptDialog = jest.fn(noopAction);

export const openConfirmationDialog = jest.fn(noopAction);

export const openGuestProductSelectionDialog = jest.fn(noopAction);

export const openGroupRegistrationTypeDialog = jest.fn(noopAction);

export const openGuestDetailsDialog = jest.fn(noopAction);

export const openGuestRemoveDialog = jest.fn(noopAction);

export const openContactPlannerDialog = jest.fn(noopAction);

export const openCancelRegistrationSuccessConfirmationDialog = jest.fn(noopAction);

export const openCancelRegistrationDialog = jest.fn(noopAction);

export const openShoulderDateApprovalWarningDialog = jest.fn(noopAction);

export const openTransactionInProcessingErrorDialog = jest.fn(noopAction);
