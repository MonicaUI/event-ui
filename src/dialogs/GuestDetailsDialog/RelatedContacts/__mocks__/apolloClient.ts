import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type RelatedContactsDialogMockData = {
  relatedContactsModalStyles: unknown;
  relatedContactsModalText: unknown;
};

export const getApolloClientMocks = (
  state: RelatedContactsDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { relatedContactsModalStyles = {}, relatedContactsModalText = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              guestRegistration: {
                addGuestFromRelatedContacts: {
                  style: {
                    relatedContactsModalStyles
                  },
                  relatedContactsModalText
                }
              }
            }
          }
        }
      }
    }
  ];
};
