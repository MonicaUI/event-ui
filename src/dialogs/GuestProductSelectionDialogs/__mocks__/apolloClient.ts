import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type GuestProductSelectionDialogMockData = {
  displayAdmissionItemsFees: unknown;
  displaySessionsFees: unknown;
};

export const getApolloClientMocks = (
  state: GuestProductSelectionDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { displayAdmissionItemsFees = {}, displaySessionsFees = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              registration: {
                admissionItems: {
                  display: {
                    fees: displayAdmissionItemsFees
                  }
                },
                sessions: {
                  display: {
                    fees: displaySessionsFees
                  }
                }
              }
            }
          }
        }
      }
    }
  ];
};
