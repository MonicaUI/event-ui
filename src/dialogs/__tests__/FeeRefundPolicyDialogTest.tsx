import { mount } from 'enzyme';
import { openFeeRefundPolicyDialog } from '../FeeRefundPolicyDialog';
import createStoreWithMiddleware from 'nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware';
import { combineReducers } from 'redux';
import dialogContainer from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import EventSnapshot from '../../../fixtures/EventSnapshot.json';
import { Provider } from 'react-redux';
import DialogContainer from 'nucleus-guestside-site/src/containers/DialogContainer';
// eslint-disable-next-line jest/no-mocks-import
import { getMockedMessageContainer } from '../__mocks__/documentElementMock';
import React from 'react';

jest.mock('../../redux/pathInfo', () => ({
  routeToPage: jest.fn(() => () => {})
}));
const noOp = c => c;
getMockedMessageContainer();

const store = createStoreWithMiddleware(
  combineReducers({
    dialogContainer,
    website: (x = {}) => x,
    text: (x = {}) => x
  }),
  {
    website: EventSnapshot.eventSnapshot.siteEditor.website,
    text: {
      translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx)
    }
  }
);

const feeModel = {
  translateCurrency: noOp,
  translate: noOp,
  eventCurrencyCode: 'USD',
  eventTimeZone: {
    id: 35,
    name: 'Eastern Time',
    nameResourceKey: 'Event_Timezone_Name_35__resx',
    plannerDisplayName: '(GMT-05:00) Eastern [US & Canada]',
    hasDst: true,
    utcOffset: -300,
    abbreviation: 'ET',
    abbreviationResourceKey: 'Event_Timezone_Abbr_35__resx',
    dstInfo: []
  },
  fee: {
    feeId: 'd950c1c7-06e3-4485-b7bd-35cc961238cb',
    feeName: 'Standard Rate',
    amount: 350.0,
    productId: '1945bc48-962a-480a-b6f0-f2877f90871d',
    productType: 'AdmissionItem',
    chargePolicies: [
      {
        id: 'cbee5cc7-2b1c-4856-9bec-cbbde0ff6726',
        isActive: true,
        effectiveUntil: 1571805860,
        amount: 200.0,
        maximumRefundAmount: 900.0
      },
      {
        id: 'dcee5cc7-2b1c-4856-9bec-cbbde0ff6726',
        isActive: true,
        effectiveUntil: 32503593600000,
        amount: 300.0,
        maximumRefundAmount: 800.0
      }
    ],
    refundPolicies: [
      {
        id: '58ca4381-3ce7-4836-9cb3-fe65c3f9f95a',
        isActive: true,
        refundType: 1,
        effectiveUntil: 1571805860,
        amount: 150.0,
        isBeforeCurrentDate: false
      },
      {
        id: '58ca4381-3ce7-4836-9cb3-fe65c3f9f95b',
        isActive: true,
        refundType: 1,
        effectiveUntil: 1574743460,
        amount: 100.0,
        isBeforeCurrentDate: false
      }
    ]
  }
};

describe('Fee refund policy dialog', () => {
  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openFeeRefundPolicyDialog(feeModel));
    expect(dialog).toMatchSnapshot();
  });
});
