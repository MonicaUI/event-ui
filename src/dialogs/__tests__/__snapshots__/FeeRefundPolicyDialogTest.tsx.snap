// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fee refund policy dialog matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; height: 50%; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class=""
              >
                <div
                  class="dialogChildren"
                >
                  <div>
                    <div
                      aria-describedby="dialogHeader"
                      class=""
                      style="background-color: rgb(221, 226, 230); border-radius: 0;"
                    >
                      <div
                        class=""
                        id="dialogHeader"
                        style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                      >
                        <h2
                          class=""
                          style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                        >
                          <div
                            class=""
                            data-cvent-id="close"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="container"
                            >
                              <span
                                aria-label="fallbackback"
                                class=""
                                role="img"
                              />
                            </span>
                          </div>
                          EventWidgets_FeeWidget_RefundPolicyModalHeader__resx
                        </h2>
                      </div>
                    </div>
                    <div
                      class=""
                    >
                      <div
                        class=""
                      >
                        <div>
                          <div
                            class=""
                            style="font-weight: 700;"
                          >
                            EventWidgets_FeeWidget_RefundModal_RefundBeforeDate__resx
                          </div>
                          <div
                            class=""
                            style="margin: 0.5rem 0px 1.75rem;"
                          >
                            <span>
                              EventWidgets_FeeWidget_RefundModal_PricePaid__resx
                            </span>
                          </div>
                        </div>
                        <div
                          class=""
                          style="padding: 0px 0.75rem 0.25rem 0.75rem;"
                        >
                          <span
                            class=""
                          >
                            EventWidgets_FeeWidget_RefundModal_CancelledBy__resx
                          </span>
                          <span
                            class=""
                          >
                            EventWidgets_FeeWidget_RefundModal_AmountRefunded__resx
                          </span>
                        </div>
                        <div>
                          <div
                            class=""
                          >
                            <div
                              class=""
                            >
                              _dateLabels_day_month_year__resx
                            </div>
                            <div
                              class=""
                            >
                              150
                            </div>
                          </div>
                          <div
                            class=""
                          >
                            <div
                              class=""
                            >
                              _dateLabels_day_month_year__resx
                            </div>
                            <div
                              class=""
                            >
                              100
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class=""
                      >
                        <div>
                          <div
                            class=""
                            style="font-weight: 700;"
                          >
                            EventWidgets_FeeWidget_RefundModal_RefundAfterDate__resx
                          </div>
                          <div
                            class=""
                            style="margin: 0.5rem 0px 1.75rem;"
                          >
                            <span>
                              EventWidgets_FeeWidget_RefundModal_PricePaid__resx
                            </span>
                          </div>
                        </div>
                        <div
                          class=""
                          style="padding: 0px 0.75rem 0.25rem 0.75rem;"
                        >
                          <span
                            class=""
                          >
                            EventWidgets_FeeWidget_RefundModal_CancelledBy__resx
                          </span>
                          <span
                            class=""
                          >
                            EventWidgets_FeeWidget_RefundModal_AmountRefunded__resx
                          </span>
                        </div>
                        <div>
                          <div
                            class=""
                          >
                            <div
                              class=""
                            >
                              _dateLabels_day_month_year__resx
                            </div>
                            <div
                              class=""
                            >
                              100
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;
